src/main/java/TaskList.java

```diff
@@ -7,16 +7,27 @@ public TaskList() {
         taskCount = 0;
     }
 
-    // Adds task to taskList
-    public String addTask(String taskName) {
-        if (taskName.equals("")){
-            return "Can't add a blank line to the task list!";
-        } else {
-            taskList[taskCount] = new Task(taskName);
-            taskCount++;
-            return "Added: " + taskName;
+    // Adds task to taskList using the Command object
+    public String addTask(Command command) {
+        // First argument will always be the description of the task
+        String taskName = command.getArgument(0).trim();
+        // Extra argument for Deadline and Event
+        String extraArg;
+
+        switch(command.getCommand().toLowerCase()) {
+        case "todo":
+            taskList[taskCount] = new Todo(taskName);
+            break;
+        case "deadline":
```

Comment:
Good implementation of the switch statement to handle individual states of operation!